platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/JsonFeature.java

```diff
@@ -71,19 +76,21 @@ public JsonFeature(Feature feature, FeaturesService featureService) {
    * @param version the feature optional version
    * @param description the feature optional description
    * @param state the feature state
+   * @param required whether the feature is required or not
    * @param region the feature optional region
    * @param startLevel the feature start level
    * @throws IllegalArgumentException if <code>name</code>, <code>id</code>, or <code>state</code>
    *     is <code>null</code>
    */
   @SuppressWarnings(
-      "squid:S00107" /* Pojo attributes initialization of final attributes requires 8 parameters */)
+      "squid:S00107" /* Pojo attributes initialization of final attributes requires 9 parameters */)
```

Comment:
These are only used in test cases so I actually just changed to package-private. The normal way to create these objects are via the one constructor that takes a Karaf feature or an OSGI bundle (in the JsonBundle class). 

Since it is not meant to be initialized in any other way and using the Karaf feature object or the OSGI bundle object is kinda close to a builder notion, I don't see the need to complicate this more by adding builders. 

Hope that is ok with you.